### PR TITLE
📖 add a note to check with maintainers for blocked comms emails

### PR DESCRIPTION
**Pull request by nawazkh**
**What this PR does / why we need it**:
- Adds a note to the `mailing-list` step of `#### Continuously: Communicate key dates to the community` task.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -367,6 +367,7 @@ upcoming code freezes etc based on the [release timeline (1.4 example)](./releas
 
 Information can be distributed via:
 * `sig-cluster-lifecycle` mailing list
+  * Note: Ensure that the communication email was received by the community. If unsure, check with the maintainers if the email was pending for approval and needs to be unblocked.
```
Comment (sbueringer) [inline docs/release/release-tasks.md:370]: I think only sig cluster lifecycle leadership has the rights.

@fabriziopandini ?

Comment (fabriziopandini) [reply docs/release/release-tasks.md:370]: confirmed, only a few sig leaders can do this operation (access rights for mailing lists are tricky, I know there was some work in progress but I'm not really up to date)

[the remark this answers is not in the document]
Comment (nawazkh) [reply docs/release/release-tasks.md:371]: Added :)

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -82,9 +82,10 @@ As of now we ask for volunteers in Slack and office hours.
 
 #### Finalize release schedule and team
 
-1. Finalize release schedule and team in `release-1.4.md`.
+1. Finalize release schedule and team in `release-x.y.md`.
```
Comment (sbueringer) [inline docs/release/release-tasks.md:85]: I don't mind but just fyi "* The examples in this document are based on the v1.4 release cycle."

The idea was to have concrete examples to make everything easier to understand.

I"m fine either way, but we should probably be consistent and either use a concrete example or use x.y, x.y+1, x.y-1 etc

File: docs/release/release-tasks.md (hunk 1 above)
Comment (sbueringer) [reply docs/release/release-tasks.md:371]: As per our understand folks have to be just members of the mailing list. I would keep this simple and just say whoever wants to send mails should join the mailing list before.

Then they can send mails, and if they see that the mail doesn't show up on the mailing list they can ask sig leadership / Fabrizio to check if the mail has to be approved

Comment (fabriziopandini) [reply docs/release/release-tasks.md:371]: member in this case means subscribed at the mailing list, sorry for the confusion

File: docs/release/release-tasks.md (hunk 2 above)
Comment (nawazkh) [reply docs/release/release-tasks.md]: Makes sense to me. Updated.

File: docs/release/release-tasks.md (hunk 1 above)
Comment (nawazkh) [reply docs/release/release-tasks.md]: Makes sense to me. Updated.

**Comment by sbueringer** [comment]
Let's go ahead with this one

/lgtm
/approve

**Pull request merged**